// @backstage/theme/dist/index.esm.js
import { createMuiTheme } from '@material-ui/core';
import { darken, lighten } from '@material-ui/core/styles/colorManipulator';
import { yellow } from '@material-ui/core/colors';

const DEFAULT_FONT_FAMILY = '"Helvetica Neue", Helvetica, Roboto, Arial, sans-serif';
function createThemeOptions(options) {
  const {palette, fontFamily = DEFAULT_FONT_FAMILY} = options;
  return {
    palette,
    props: {
      MuiGrid: {
        spacing: 2
      },
      MuiSwitch: {
        color: "primary"
      }
    },
    typography: {
      fontFamily,
      h5: {
        fontWeight: 700
      },
      h4: {
        fontWeight: 700,
        fontSize: 28,
        marginBottom: 6
      },
      h3: {
        fontSize: 32,
        fontWeight: 700,
        marginBottom: 6
      },
      h2: {
        fontSize: 40,
        fontWeight: 700,
        marginBottom: 8
      },
      h1: {
        fontSize: 54,
        fontWeight: 700,
        marginBottom: 10
      }
    }
  };
}
function createThemeOverrides(theme) {
  return {
    MuiCssBaseline: {
      "@global": {
        html: {
          height: "100%",
          fontFamily: theme.typography.fontFamily
        },
        body: {
          height: "100%",
          fontFamily: theme.typography.fontFamily,
          "overscroll-behavior-y": "none"
        },
        a: {
          color: "inherit",
          textDecoration: "none"
        }
      }
    },
    MuiTableRow: {
      root: {
        "&:nth-of-type(odd)": {
          backgroundColor: theme.palette.background.default
        }
      },
      hover: {
        "&:hover": {
          cursor: "pointer"
        }
      },
      head: {
        "&:nth-of-type(odd)": {
          backgroundColor: theme.palette.background.paper
        }
      }
    },
    MuiTableCell: {
      root: {
        wordBreak: "break-word",
        overflow: "hidden",
        verticalAlign: "middle",
        lineHeight: "1",
        margin: 0,
        padding: theme.spacing(3, 2, 3, 2.5),
        borderBottom: 0
      },
      sizeSmall: {
        padding: theme.spacing(1, 2, 1, 2.5)
      },
      head: {
        wordBreak: "break-word",
        overflow: "hidden",
        color: "rgb(179, 179, 179)",
        fontWeight: "normal",
        lineHeight: "1"
      }
    },
    MuiTabs: {
      root: {
        minHeight: 24
      }
    },
    MuiTab: {
      root: {
        color: theme.palette.link,
        minHeight: 24,
        textTransform: "initial",
        letterSpacing: "0.07em",
        "&:hover": {
          color: darken(theme.palette.link, 0.3),
          background: lighten(theme.palette.link, 0.95)
        },
        [theme.breakpoints.up("md")]: {
          minWidth: 120,
          fontSize: theme.typography.pxToRem(14),
          fontWeight: 500
        }
      },
      textColorPrimary: {
        color: theme.palette.link
      }
    },
    MuiTableSortLabel: {
      root: {
        color: "inherit",
        "&:hover": {
          color: "inherit"
        },
        "&:focus": {
          color: "inherit"
        }
      },
      active: {
        fontWeight: "bold",
        color: "inherit"
      }
    },
    MuiListItemText: {
      dense: {
        whiteSpace: "nowrap",
        overflow: "hidden",
        textOverflow: "ellipsis"
      }
    },
    MuiButton: {
      text: {
        padding: void 0
      }
    },
    MuiChip: {
      root: {
        backgroundColor: "#D9D9D9",
        marginRight: theme.spacing(1),
        marginBottom: theme.spacing(1)
      },
      label: {
        color: theme.palette.grey[900],
        lineHeight: `${theme.spacing(2.5)}px`,
        fontWeight: theme.typography.fontWeightMedium,
        fontSize: `${theme.spacing(1.75)}px`
      },
      labelSmall: {
        fontSize: `${theme.spacing(1.5)}px`
      },
      deleteIcon: {
        color: theme.palette.grey[500],
        width: `${theme.spacing(3)}px`,
        height: `${theme.spacing(3)}px`,
        margin: `0 ${theme.spacing(0.75)}px 0 -${theme.spacing(0.75)}px`
      },
      deleteIconSmall: {
        width: `${theme.spacing(2)}px`,
        height: `${theme.spacing(2)}px`,
        margin: `0 ${theme.spacing(0.5)}px 0 -${theme.spacing(0.5)}px`
      }
    },
    MuiCardHeader: {
      root: {
        paddingBottom: 0
      }
    },
    MuiCardActions: {
      root: {
        justifyContent: "flex-end"
      }
    }
  };
}
function createTheme(options) {
  const themeOptions = createThemeOptions(options);
  const baseTheme = createMuiTheme(themeOptions);
  const overrides2 = createThemeOverrides(baseTheme);
  const theme = {...baseTheme, overrides: overrides2};
  return theme;
}

const lightTheme = createTheme({
  palette: {
    type: "light",
    background: {
      default: "#F8F8F8"
    },
    status: {
      ok: "#1DB954",
      warning: "#FF9800",
      error: "#E22134",
      running: "#2E77D0",
      pending: "#FFED51",
      aborted: "#757575"
    },
    bursts: {
      fontColor: "#FEFEFE",
      slackChannelText: "#ddd",
      backgroundColor: {
        default: "#7C3699"
      }
    },
    primary: {
      main: "#2E77D0"
    },
    banner: {
      info: "#2E77D0",
      error: "#E22134",
      text: "#FFFFFF",
      link: "#000000"
    },
    border: "#E6E6E6",
    textContrast: "#000000",
    textVerySubtle: "#DDD",
    textSubtle: "#6E6E6E",
    highlight: "#FFFBCC",
    errorBackground: "#FFEBEE",
    warningBackground: "#F59B23",
    infoBackground: "#ebf5ff",
    errorText: "#CA001B",
    infoText: "#004e8a",
    warningText: "#FEFEFE",
    linkHover: "#2196F3",
    link: "#0A6EBE",
    gold: yellow.A700,
    navigation: {
      background: "#171717",
      indicator: "#9BF0E1"
    },
    pinSidebarButton: {
      icon: "#181818",
      background: "#BDBDBD"
    },
    tabbar: {
      indicator: "#9BF0E1"
    }
  }
});
const darkTheme = createTheme({
  palette: {
    type: "dark",
    background: {
      default: "#333333"
    },
    status: {
      ok: "#71CF88",
      warning: "#FFB84D",
      error: "#F84C55",
      running: "#3488E3",
      pending: "#FEF071",
      aborted: "#9E9E9E"
    },
    bursts: {
      fontColor: "#FEFEFE",
      slackChannelText: "#ddd",
      backgroundColor: {
        default: "#7C3699"
      }
    },
    primary: {
      main: "#2E77D0"
    },
    banner: {
      info: "#2E77D0",
      error: "#E22134",
      text: "#FFFFFF",
      link: "#000000"
    },
    border: "#E6E6E6",
    textContrast: "#FFFFFF",
    textVerySubtle: "#727272",
    textSubtle: "#CCCCCC",
    highlight: "#FFFBCC",
    errorBackground: "#FFEBEE",
    warningBackground: "#F59B23",
    infoBackground: "#ebf5ff",
    errorText: "#CA001B",
    infoText: "#004e8a",
    warningText: "#FEFEFE",
    linkHover: "#2196F3",
    link: "#0A6EBE",
    gold: yellow.A700,
    navigation: {
      background: "#424242",
      indicator: "#9BF0E1"
    },
    pinSidebarButton: {
      icon: "#404040",
      background: "#BDBDBD"
    },
    tabbar: {
      indicator: "#9BF0E1"
    }
  }
});

export { createTheme, createThemeOptions, createThemeOverrides, darkTheme, lightTheme };
